pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +321,426 @@ func TestMemberService(t *testing.T) {
 		})
 	}
 }
+
+func TestStatefulSetForRack(t *testing.T) {
+	basicRack := scyllav1.RackSpec{
+		Name: "rack",
+		Storage: scyllav1.StorageSpec{
+			Capacity: "1Gi",
+		},
+	}
+
+	newBasicExistingStatefulSet := func() *appsv1.StatefulSet {
+		return &appsv1.StatefulSet{}
+	}
+
+	basicSidecarImage := ""
+
+	newBasicScyllaCluster := func() *scyllav1.ScyllaCluster {
+		sc := &scyllav1.ScyllaCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "basic",
+				UID:  "the-uid",
+			},
+			Spec: scyllav1.ScyllaClusterSpec{
+				Datacenter: scyllav1.DatacenterSpec{
+					Name: "dc",
+					Racks: []scyllav1.RackSpec{
+						basicRack,
+					},
+				},
+			},
+			Status: scyllav1.ScyllaClusterStatus{
+				Racks: map[string]scyllav1.RackStatus{},
+			},
+		}
+		return sc
+	}
+
+	newBasicStatefulSetLabels := func() map[string]string {
+		return map[string]string{
+			"app":                          "scylla",
+			"app.kubernetes.io/managed-by": "scylla-operator",
+			"app.kubernetes.io/name":       "scylla",
+			"scylla/cluster":               "basic",
+			"scylla/datacenter":            "dc",
+			"scylla/rack":                  "rack",
+		}
+	}
+
+	newBasicStatefulSetLabelsWithVersion := func() map[string]string {
+		m := newBasicStatefulSetLabels()
+		m["scylla/scylla-version"] = ""
+		return m
+	}
+
+	newBasicExpectedStatefulSet := func() *appsv1.StatefulSet {
```

Comment:
this doesn't need the "expected" part, you'd use it in both the existing and expected parts